### ISSUE TITLE
isAdvancedPricingModel Inconsistent

### ISSUE THREAD

**Issue by adamhewitt627** [microsoft/StoreBroker#129]
### Environment
- ModuleVersion 1.18.1
- PSVersion 5.1.17134.228
- Windows 10 - 1803

### Summary
Again, this may be more of an issue on the Store backend (Fiddler suggests that StoreBroker isn't interfering) but I am having trouble with getting `InAppProductSubmissions` to succeed. All I am attempting to change is pricing from Free to a TierX, now that our account is configured with a payout. Sometimes they stick on `CommitStarted`, though I also got "Submission has active validation errors which cannot be exposed via API, please visit https://developer.microsoft.com to see the issue." (No details were available on the dashboard)

Based on inspections from Fiddler, it _seems_ to be an issue with isAdvancedPricingModel. Creating a new submission, it is `$true`, but the Store returns `$false` when I `PUT` the changes. (I _am_ able to use the advanced tiers if I create/edit/commit the submission within the portal, and browsing the tiers within the portal only shows the advanced set)

### Code
```PowerShell
$submission = New-InAppProductSubmission -IapId $IapId -Force
Set-InAppProductSubmission -IapId $IapId -UpdatedSubmission @{
    id = $submission.id
    pricing = @{
        isAdvancedPricingModel = $true
        priceId = "Tier1152"
    }
}
```

Results in a request:
```http
PUT https://manage.devcenter.microsoft.com/v1.0/my/inappproducts/.../submissions/... HTTP/1.1

{ 
  "id":"...",
  "pricing":  {
    "isAdvancedPricingModel":  true,
    "priceId":  "Tier1152"
  }
}
```

and response:
```
HTTP/1.1 200 OK
Content-Length: 1088
Content-Type: application/json; charset=utf-8
MS-CorrelationId: 373dc845-b94b-4608-9535-257842daae67
MS-RequestId: ed6e6429-ab2f-4921-88ab-921319fdef33
MS-CV: f+l06wOq2EGbiCO+.8
MS-ServerId: 101112022
Date: Tue, 28 Aug 2018 19:52:17 GMT

{"id":"...","pricing":{"marketSpecificPricings":{},"sales":[],"priceId":"Tier1152","isAdvancedPricingModel":false}}
```

### Expected
`isAdvancedPricingModel` shouldn't change with the `PUT`, especially since documentation says it is read only.

--------------------------------------
**Other small question that doesn't seem worth its own issue:** I have thousands of in-app products that I'm working toward processing, and will need continued updates with price changes. I'm seeing ~30-60 seconds each, mostly spent waiting on the server in `Cloning current submission`. Is there a better way to batch this work that I'm missing? My understanding is the REST API is the official way, but it's so slow.

**Comment by danbelcher-MSFT** [comment]
Regarding the question about time spent cloning the current submission, see my question and Howard's response from one of the recent PRs in which support was added for `isMinimalResponse`: https://github.com/Microsoft/StoreBroker/pull/124

**Comment by adamhewitt627** [comment]
How does `isMinimalResponse` apply here, since it specifically says it _does not_ apply to IAPs?

**Comment by adamhewitt627** [comment]
I contacted dev center support, and was told the IAP submission failure is a result of not answering a new privacy policy question. I tried StoreBroker on a product that had the question answered, and found:
1. I saw the same behavior as reported above (`isAdvancedPricingModel` coming back false after `Set-InAppProductSubmission`)
2. The product was able to pass certification, despite appearing to have the wrong pricing tier.

**Conclusion**: it doesn't appear that there is an issue with StoreBroker. However, I'm still waiting to hear from support if there is a way to answer the question through the API. We have over 10k IAPs that will need the flag set.

**Comment by adamhewitt627** [comment]
Check this page in a browser. I ended up having to write a Selenium script to loop through and mark the policy as not-needed.
```
"https://partner.microsoft.com/en-us/dashboard/products/{iapId}/submissions/{submissionId}/properties
```

**Comment by johanlindfors** [comment]
That is just really bad, not having those properties exposed in the API that is... As of now, I do not want to include a second "automation" technology to get this working, will instead try and leverage some channels and suggest a rapid update on the API.

**Comment by adamhewitt627** [comment]
I agree, and hope you can make progress. If it helps, it is only necessary _once_ per add-on, as part of its first submission. After that, price/listing updates can be driven entirely with the REST API.

**Comment by HowardWolosky** [comment]
I've discussed this issue with the API team in the past (and again quite recently).  At this time, it is not in their plan to expose the "form" type properties through the API.

**Comment by johanlindfors** [comment]
Appreciate the effort @HowardWolosky. I will try an approach as well! I have 21 IAP's that I was ready to submit, and I'm not enthustiastic about the manual process for all of these tomorrow... I know there's no way this will change over the night, but one has to try...

**Comment by HowardWolosky** [comment]
Closing this out as there's nothing more that can be done on our side right now, and the feedback has been provided to the team that owns the API.
